modin/experimental/engines/omnisci_on_ray/io.py

```diff
@@ -288,3 +338,94 @@ def _prepare_pyarrow_usecols(cls, read_csv_kwargs):
                 raise NotImplementedError("unsupported `usecols` parameter")
 
         return usecols_md
+
+    @classmethod
+    def _read_csv_check_support(
+        cls,
+        read_csv_kwargs: ReadCsvKwargsType,
+    ) -> Tuple[bool, str]:
+        """
+        Check whatever or not passed parameters are supported by current modin.read_csv
+        implementation.
+
+        Parameters
+        ----------
+        read_csv_kwargs: ReadCsvKwargsType
+                Parameters of read_csv function.
+
+        Returns
+        -------
+        bool:
+            Whatever passed parameters are supported or not.
+        str:
+            Error message that should be raised if user explicitely set `engine="arrow"`.
+        """
+        filepath_or_buffer = read_csv_kwargs.get("filepath_or_buffer", None)
+        header = read_csv_kwargs.get("header", "infer")
+        names = read_csv_kwargs.get("names", None)
+        engine = read_csv_kwargs.get("engine", None)
+        skiprows = read_csv_kwargs.get("skiprows", None)
+        delimiter = read_csv_kwargs.get("delimiter", None)
+
+        if isinstance(filepath_or_buffer, str):
+            file_extension = os.path.splitext(filepath_or_buffer)[1]
+            if (
+                not cls.file_exists(filepath_or_buffer, only_locally=True)
+                or file_extension != ".csv"
+            ):
+                return False, cls.not_implemented_message
```

Comment:
If a file doesn't exist, we should provide a user with a better error message.

Reply:
Ok, more specific comments were added.

Reply:
What if a user explicitly chooses Arrow engine and provides a wrong file name? Will he get NotImplementedError?

Reply:
Yes, he will. To avoid this confusion, i have added `ArrowEngineException` to raise it instead of `NotImplementedError` exception.

Reply:
Raising the same error for all issues is not very convenient. `FileNotFoundError` would be much more appropriate here and this would match Pandas' behavior. How about raising various errors right here instead of returning messages?

Reply:
Ok, i think it is good idea. Exceptions were added in the `_read_csv_check_support` function.